f3/powertable.go

```diff
@@ -1,22 +1,75 @@
 package f3
 
-// A power table maps participant IDs to power values.
+import (
+	"sort"
+)
+
+// PowerEntry represents a single entry in the PowerTable.
+// It includes an ActorID and its Weight
+type PowerEntry struct {
+	ID     ActorID
+	Power  *StoragePower
+	PubKey []byte
+}
+
+// PowerTable maps ActorID to a unique index in the range [0, len(powerTable.Entries)).
+// Entries is the reverse mapping to a PowerEntry.
 type PowerTable struct {
-	Entries map[ActorID]uint
-	Total   uint
+	Entries []PowerEntry    // Slice to maintain the order. Meant to be maintained in order in order by (Power descending, ID ascending)
+	Lookup  map[ActorID]int // Maps ActorID to the index of the associated entry in Entries
+	Total   *StoragePower
 }
 
-func NewPowerTable() PowerTable {
-	return PowerTable{
-		Entries: map[ActorID]uint{},
-		Total:   0,
+// NewPowerTable creates a new PowerTable from a slice of PowerEntry .
+// It is more efficient than Add, as it only needs to sort the entries once.
+func NewPowerTable(entries []PowerEntry) *PowerTable {
```

Comment:
Please note the function takes ownership of the slice - it must not be modified afterwards.

Reply:
Correct, but not yet an issue. I reckon we leave as is?

Reply:
Sorry, by note I meant "write a comment"

Reply:
Done